Snakefile

```diff
@@ -25,207 +25,229 @@ f.close()
 #get seeds and hemis from config
 seeds = config['seeds']
 hemis = config['hemis']
-
+excrois = config['excrois']
 
 wildcard_constraints:
-    subject="[a-zA-Z0-9]+",
-    template="[a-zA-Z0-9]+"
+subject="[a-zA-Z0-9]+",
+template="[a-zA-Z0-9]+"
 
 
 rule all:
-    input: 
-        clusters = expand('diffparc/clustering/group_space-{template}_seed-{seed}_hemi-{hemi}_method-spectralcosine_k-{k}_cluslabels.nii.gz',seed=seeds,hemi=hemis,template=config['template'],k=range(2,config['max_k']+1))
+	input: 
+		clusters = expand('diffparc/clustering/group_space-{template}_seed-{seed}_hemi-{hemi}_method-spectralcosine_k-{k}_cluslabels.nii.gz',seed=seeds,hemi=hemis,template=config['template'],k=range(2,config['max_k']+1))
 
 
 
 rule import_targets:
-    input: 
-        in_seg = join(config['targets_seg_dir'],config['targets_seg_bilat']),
-        lut_in = config['targets_lut_in'],
-        lut_out = config['targets_lut_out']
-    output: 
-        out_seg = 'diffparc/sub-{subject}/masks/lh_rh_targets_native.nii.gz'
-    envmodules: 'mrtrix'
-    singularity: config['singularity_neuroglia']
-    log: 'logs/import_targets_hcp_mmp_sym/sub-{subject}.log'
-    group: 'pre_track'
-    shell:
-        'labelconvert  {input.in_seg} {input.lut_in} {input.lut_out} {output.out_seg} &> {log}'
+	input: 
+		in_seg = join(config['targets_seg_dir'],config['targets_seg_bilat']),
+		lut_in = config['targets_lut_in'],
+		lut_out = config['targets_lut_out']
+	output: 
+		out_seg = 'diffparc/sub-{subject}/masks/lh_rh_targets_native.nii.gz'
+	envmodules: 'mrtrix'
+	singularity: config['singularity_neuroglia']
+	log: 'logs/import_targets_hcp_mmp_sym/sub-{subject}.log'
+	group: 'pre_track'
+	shell:
+		'labelconvert  {input.in_seg} {input.lut_in} {input.lut_out} {output.out_seg} &> {log}'
 
 
 rule import_template_seed:
-    input: join(config['template_seg_dir'],config['template_seg_nii'])
-    output: 'diffparc/template_masks/sub-{template}_hemi-{hemi}_desc-{seed}_mask.nii.gz'
-    log: 'logs/import_template_seed/{template}_{seed}_{hemi}.log'
-    group: 'pre_track'
-    shell: 'cp -v {input} {output} &> {log}'
+	input: join(config['template_seg_dir'],config['template_seg_nii'])
+	output: 'diffparc/template_masks/sub-{template}_hemi-{hemi}_desc-{seed}_mask.nii.gz'
+	log: 'logs/import_template_seed/{template}_{seed}_{hemi}.log'
+	group: 'pre_track'
+	shell: 'cp -v {input} {output} &> {log}'
 
 
 rule import_seed_subject:
-    input: 
-        seed_nii = join(config['seed_seg_dir'],config['seed_seg_nii'])
-    output:
-        seed_nii = 'diffparc/sub-{subject}/masks/seed_{seed}_{hemi}.nii.gz'
-    shell: 'cp -v {input} {output}'
- 
+	input: 
+		seed_nii = join(config['seed_seg_dir'],config['seed_seg_nii'])
+	output:
+		seed_nii = 'diffparc/sub-{subject}/masks/seed_{seed}_{hemi}.nii.gz'
+	shell: 'cp -v {input} {output}'
+
+rule import_excrois_subject:
```

Comment:
since this rule is effectively taking all the excrois and combining into a single image, you don't want to have a {excroi} wildcard (since that would invoke a job for each excroi).. instead you want to use the `expand()` function to loop over excrois in the inputs

Reply:
I've made a commit just now with that suggestion, have a look and let me know what you think

Reply:
for the shell command, an easier strategy could be something like:
```
fslmerge -t {output.combined_4d} {input.excroi_nii} &&
fslmaths {output.combined_4d} -Tmax {output.com_excrois}
```

Reply:
> since this rule is effectively taking all the excrois and combining into a single image, you don't want to have a {excroi} wildcard (since that would invoke a job for each excroi).. instead you want to use the expand() function to loop over excrois in the inputs

is there a way to use expand() and just take the excrois and not the subjects? I tried to use expand() but it takes all the subjects as well (which it should) but then I am concerned if we merge all excrois over the subjects would it not interfere with protract? shouldn't the --avoid {exclusion mask} be applied for each subject in their native space?

Reply:
yes, the way to do exactly that is to use the `allow_missing=True` option, that basically lets you skip other wildcards in the expand statement  (i.e. it will keep {subject} as a wildcard)

Reply:
a few simple examples to illustrate: 
if we had: 
```
subjects=['1','2']
excroi=['a','b']
```
This one:
```
expand('{subject}_{excroi}.nii.gz', subject=subjects,excroi=excrois)
```
would evaluate to:
`['1_a.nii.gz', '1_b.nii.gz', '2_a.nii.gz', '2_b.nii.gz']`

but this:
```
expand('{subject}_{excroi}.nii.gz',excroi=excrois, allow_missing=True)
```
would evaluate to:
`['{subject}_a.nii.gz', '{subject}_b.nii.gz']`